### ISSUE TITLE
Outdated information about memory limits

### ISSUE THREAD

**Issue by mkuratczyk** [docker-library/rabbitmq#621]
The `Memory Limits` section of https://hub.docker.com/_/rabbitmq is incorrect (perhaps was true a long time ago). Container limits are taken into account, so `vm_memory_high_watermark.relative` will be calculated based on container memory, not host memory. I think the whole section can be simply removed.

**Comment by mkuratczyk** [comment]
Turns out this can still happen. A user created an issue today (using up to date version of RabbitMQ and Erlang) and I noticed:
```
Memory high watermark setting: 0.4 of available memory, computed to: 108.0799 gb
```
I'm fairly sure this is the host's memory, not container memory limit. I wonder what could be causing this in some situations.

**Comment by michaelklishin** [comment]
Erlang's imperfect detection of the limits using cgroups? cgroup version differences (IIRC some distribution have started rolling out v2 recently)?

**Comment by Zerpet** [comment]
In rabbitmq cluster operator, we set `total_memory_available_override_value` to 80% of the resource limit in the Pod i.e. limit by cgroup. We could document this setting in the `Memory Limits` section, if Erlang doesn't play well with cgroups.

**Comment by michaelklishin** [comment]
@luarx no, it never did work reliably because the runtime's ability to detect the amount of available memory is only limited to some OSes and likely cgroups v1 for now.

The docs have been [updated to provide Kubernetes-specific recommendations](https://www.rabbitmq.com/docs/memory#configuring-threshold).

**Comment by michaelklishin** [comment]
@tianon @yosifkit folks, can we integrate [these recommendations from the RabbitMQ docs](https://www.rabbitmq.com/docs/memory#configuring-threshold) into this image, and then hopefully consider this issue resolved?

I would be happy to contribute if you tell me where to look in the source, and how to test a preview of my edits locally. Thank you!

**Comment by tianon** [comment]
That's in https://github.com/docker-library/docs/blob/master/rabbitmq/content.md :eyes: (as linked above :sweat_smile:)

If you want to test the Docker Hub rendering, you can just create a repository and there's a preview in the private "edit" page, but I think markdown is probably limited enough that you can probably guess how it'll render pretty well (Hub's rendering is pretty vanilla). :innocent:

**Comment by JefSeys** [comment]
Any chance there is an update on this issue? I'd like to set the free disk space but this is not supported atm AFAIK.

**Comment by michaelklishin** [comment]
@JefSeys this issue has nothing to do with free disk space limits or how this image approaches configuration.

You can [set the free disk space alarm](https://www.rabbitmq.com/docs/disk-alarms#configure) threshold [using `rabbitmq.conf`](https://www.rabbitmq.com/docs/configure), just like 99% of other settings that can be set there. Unlike environment variables, it offers validation.

Any further comments or questions about free disk space limits will be deleted. This repository has discussions enabled, please start a new one instead of commenting on an open issue that is about a different limit in a specific context.

**Comment by LaurentGoderre** [comment]
I would argue that the docs as they are now are sufficient. It provides a brief overview and a link to an authoritative source for more details.
